discovery/README.md

```diff
@@ -216,15 +216,62 @@ If all the targets in a group go away, we need to send the target groups with em
 ```
 down the channel.
 
+### The Config interface
+
+Now that your service discovery mechanism is ready to discover targets, you must help
+Prometheus discover it. This is done my implementing the `discovery.Config` interface
+and registering it with `config.RegisterServiceDiscovery` in an init function of your
+package.
+
+```go
+type Config interface {
+	// Name returns the name of the discovery mechanism.
+	Name() string
+
+	// NewDiscoverer returns a Discoverer for the Config
+	// with the given DiscovererOptions.
+	NewDiscoverer(DiscovererOptions) (Discoverer, error)
+
+	// SetOptions applies the ConfigOptions to the Config.
+	SetOptions(ConfigOptions)
+
+	// Validate checks the Config for errors.
+	Validate() error
+}
+
+type ConfigOptions struct {
+	// Directory may be used to resolve relative file paths
+	// in the config (e.g. TLS certificates).
+	Directory string
+}
+
+type DiscovererOptions struct {
+	Logger log.Logger
+}
+```
+
+The value returned by `Name()` should be short, descriptive, lowercase, and unique.
+It's used to tag the provided `Logger` and as the part of the YAML key for your SD
+mechanism's list of configs in `scrape_config` and `alertmanager_config`
+(e.g. `${NAME}_sd_configs`).
+
+`Validate()` is used by `promtool` to help discover configuration errors. It should
```

Comment:
should we call that outside of promtool as well then? or specify that it is expected that the SD calls it on unmarshalling? WDYT?

Reply:
Yes, this should be part of unmarshalling.

Reply:
It might be able to share some code with unmarshalling validation (e.g. checking required or "oneof" fields), but this validation also checks for the existence of files whose paths may still be relative at unmarshalling time.

Reply:
```go
func (c *Config) validate() error {
  // check for field validity (e.g. required, "oneof", "enums", etc.)
}

func (c *Config) Validate() error {
  if err := c.validate(); err != nil { return err }
  // check for file existence and possibly parse them for validity
}

func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
  type plain Config
  v := DefaultConfig
  if err := unmarshal(&v); err != nil { return err }
  if err := (*Config)(&v).validate(); err != nil { return err }
  *c = Config(*v)
  return nil
}
```

Reply:
> checks for the existence of files whose paths may still be relative at unmarshalling time.

How is that a problem? Stat works with relative paths.

Reply:
Relative to the config file.